### PR TITLE
Cart context

### PR DESCRIPTION
**Pull request by Varu98**

### CODE REVIEW

File: src/frontend/CartPage/CartPage.jsx [hunk 1]
```diff
@@ -1,8 +1,57 @@
...
+    <>
+      <div className="cart-items-wrapper">
+        {cartItems.map((cartItem) => {
+          const { _id, img, title, price, discount, ratings, desc } = cartItem;
+
+          return (
+            <div key={_id} className="price-card">
+              <AiOutlineHeart className="wishlist-icon" />
+              <span className="price-card-img">
+                <img src={img} alt="" />
```
Comment (msdeshmukh009) [inline src/frontend/CartPage/CartPage.jsx:20]: We should give a proper `alt` attribute every time cause that helps with accessibility. In this case, you can give `title` as an `alt` attribute. 
```javascript
<img src={img} alt={title} />
```

File: src/frontend/CartPage/CartPage.jsx [hunk 2]
```diff
@@ -1,8 +1,57 @@
...
+                    {" "}
+                    discount{" "}
+                  </span>
+                </span>
+              </div>
+              <div className="price-card-body">{desc}</div>
+              <div className="price-card-footer">
+                <span className="price-card-product-price">{ratings}⭐️</span>
+                <button
+                  style={{ backgroundColor: "#A30000" }}
```
Comment (msdeshmukh009) [inline src/frontend/CartPage/CartPage.jsx:37]: Avoid inline style I guess

File: src/frontend/ProductCard/ProductCard.jsx [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
 import { useProducts } from "../Contexts/ProductContext/ProductContext";
 import "./ProductCard.css";
 
 const ProductCard = () => {
   const { newProducts } = useProducts();
-  console.log(newProducts);
+  const { cartState, cartDispatch } = useCart();
+  useEffect(() => {
+    console.log(cartState);
+  }, [cartState]);
```
Comment (msdeshmukh009) [inline src/frontend/ProductCard/ProductCard.jsx:14]: I think you forgot to remove this `console.log`

File: src/frontend/ProductCard/ProductCard.jsx [hunk 4]
```diff
@@ -26,20 +33,40 @@ const ProductCard = () => {
...
             <div className="price-card-footer">
-              <button className="secondary-btn">Details</button>
-              <button className="primary-btn">
-                Buy
-                <span className="material-icons price-card-icon-cart">
-                  shopping_cart
-                </span>
-              </button>
+              <span className="price-card-product-price">{ratings}⭐️</span>
+              {cartState.cartItems.some((product) => product._id === _id) ? (
```
Comment (msdeshmukh009) [inline src/frontend/ProductCard/ProductCard.jsx:39]: We can extract this operation's result in some variable and then use it here as required. 
```javascript
const IsPresentInCart = cartState.cartItems.some((product) => product._id === _id
```

File: src/frontend/Reducers/cartReducer.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+export const cartInitialState = {
+  cartItems: [],
+};
+
+export const cartReducer = (cartState, cartAction) => {
+  switch (cartAction.type) {
+    case "ADD_TO_CART":
```
Comment (msdeshmukh009) [inline src/frontend/Reducers/cartReducer.js:7]: Try to use variables as action types, you will avoid mistakes in that way.

**Review by msdeshmukh009 — APPROVED** [review]
The code looks fine. It would have been better if we could see the deploy preview. It seems you have some unused variables and that's why netlify is failing the deployment. So maybe try fixing that as well.

**Review by Chakravarthi0 — COMMENTED** [review]
apart from @msdeshmukh009's suggestion, everything looks good

**Pull request merged**